common/browser_test.go

```diff
@@ -21,7 +21,7 @@ func TestBrowserNewPageInContext(t *testing.T) {
 	}
 	newTestCase := func(id cdp.BrowserContextID) *testCase {
 		ctx := context.Background()
-		b := newBrowser(ctx, nil, nil, NewLaunchOptions(), NewNullLogger())
+		b := newBrowser(ctx, nil, nil, NewLaunchOptions(), NewNullLogger(), nil)
```

Comment:
I think the cleanup func should be a field on the browser instead of enlarging the already large input parameters list. WDYT?

Reply:
Agreed, it does seem a bit too long. I think this and a few other files need to be abstracted and refactored a bit 😅 

I think for now I'd prefer to leave it as an input param on `NewBrowser` and `newBrowser` since to me the `cleanupFunc` is a required field (even though there's a check for `nil` if it wasn't assigned, and not really needed in tests but only during normal runtime).

If you have a reason why it shouldn't be a field param, I'd be happy to change it though.